### PR TITLE
fix(tldraw): let frame selection wrap a single shape

### PR DESCRIPTION
**Pull request by m31-galaxy**
This PR fixes a bug where the **Frame selection** action (`Cmd/Ctrl+Alt+G`, and the Edit submenu item) silently did nothing when a single shape was selected, even though the shortcuts dialog and the frame docs describe it as wrapping the current selection.

### Before

The action, added in #8151 for #6949, was modelled on the Cmd+G group/ungroup toggle and inherited its two-shape minimum: the action returned early below two selected shapes, and `FrameSelectionMenuItem` reused `useAllowGroup()`, which also requires two. A group of one is meaningless, but a frame around one shape is a real operation, and Figma's equivalent shortcut frames a single layer. Nothing in the PR, the original request, or the tests decided the single-shape case; it fell out of copying the group gate.

Opening the single-shape path also exposed two edges the minimum had been hiding from the UI. Framing a lone arrow bound to an unselected shape produced an empty frame, because `ArrowBindingUtil` reparents such an arrow straight back out. Framing a locked shape moved it by the frame padding, because `reparentShapes` ignores locks while `fitFrameToContent` skips locked children when it shifts the frame's contents. The locked case was already reachable with two or more shapes selected.

### After

The action frames any non-empty selection. A new internal helper, `getFrameableShapeIds`, drops locked shapes (matching `groupShapes`) and arrows bound to a shape that will not be in the frame, whether that shape is unselected or selected but locked; the action bails when nothing frameable remains, and only frameable shapes are reparented into the new frame. `FrameSelectionMenuItem` gets its own gate: hidden when nothing is selected, when nothing frameable remains, or when every selected shape is a frame, where the action would unframe instead and **Remove frame** already applies. That last rule also removes a mislabelled entry: selecting two frames used to show **Frame selection**, and choosing it removed them.

### Implementation notes

The menu gate no longer inherits the `useAllowGroup` rule that refuses any arrow bound outside the selection. The shortcut never enforced that rule, and the binding util keeps such an arrow out of the frame anyway, so the menu now shows the item exactly when the action would produce a non-empty frame. The frame-only check uses `isShapeOfType(shape, 'frame')` to match the action's own unframe branch rather than the `isShapeFrameLike` check in `RemoveFrameMenuItem`; that pre-existing mismatch is left alone.

`getFrameableShapeIds` drops locked shapes first and checks each arrow's bound target against what remains, not against the selection. Checking against the selection let an arrow bound to a selected but locked shape through: the action reparented only the arrow, `reparentArrow` moved it straight back to the page, and an empty frame was left selected. Arrows can't bind to arrows, so one pass is enough. The new frame's initial bounds come from `editor.getShapesPageBounds` over the frameable ids, which already returns `null` for an empty list. That placeholder frame is created before `fitFrameToContent` sizes it, and frame props reject zero dimensions, so it is clamped with `Box.ZeroFix`: a lone horizontal arrow, a flat line, or a dot used to throw a validation error at that point.

### Change type

- [x] `bugfix`

### Test plan

1. Select one shape and press `Cmd/Ctrl+Alt+G`: it is wrapped in a frame and the frame is selected. Press again to remove the frame.
2. Right-click one shape: **Edit** > **Frame selection** appears. Right-click a frame: only **Remove frame** appears.
3. Select a locked shape together with an unlocked one and frame them: only the unlocked shape moves into the frame.
4. Select an arrow bound to an unselected shape and press the shortcut: nothing happens.
5. Select a single horizontal arrow and press the shortcut: it is framed rather than throwing.

- [x] Unit tests — the shortcut tests for a single shape, a horizontal arrow, a locked shape, and an arrow bound to a selected locked shape, and the menu item gate tests, fail on `main` and pass with this change

### Release notes

- Fix the **Frame selection** action and shortcut doing nothing when a single shape is selected, and keep locked shapes and arrows bound to shapes outside the new frame out of it.

### Code changes

| Section   | LOC change |
| --------- | ---------- |
| Core code | +43 / -10  |
| Tests     | +264 / -1  |

### CODE REVIEW

File: packages/tldraw/src/lib/ui/context/actions.tsx [hunk 1]
```diff
@@ -619,12 +619,14 @@ export function ActionsProvider({ overrides, children }: ActionsProviderProps) {
...
+					// Unlike group, a single shape can be framed, so there is no two-shape minimum.
+					const ids = getFrameableShapeIds(editor, editor.getSelectedShapeIds())
+					if (ids.length === 0) return
 
 					const shapes = compact(ids.map((id) => editor.getShape(id)))
-					const pageBounds = editor.getSelectionPageBounds()
-					if (!pageBounds) return
+					const pageBounds = Box.Common(
+						compact(shapes.map((shape) => editor.getShapePageBounds(shape)))
+					)
```
[the remark this answers is not in the document]
Comment (m31-galaxy) [reply packages/tldraw/src/lib/ui/context/actions.tsx]: Taken in 76db47a. The frameable ids always resolve to existing shapes, so the list could not actually end up empty, but `editor.getShapesPageBounds(ids)` is simpler and keeps the null guard that `main` had.

File: packages/tldraw/src/lib/utils/frames/frames.ts [hunk 2]
```diff
@@ -42,6 +43,24 @@ export function removeFrame(editor: Editor, ids: TLShapeId[]) {
...
+export function getFrameableShapeIds(editor: Editor, ids: TLShapeId[]): TLShapeId[] {
+	const selected = new Set(ids)
+	return ids.filter((id) => {
+		const shape = editor.getShape(id)
+		if (!shape || shape.isLocked) return false
+		return editor
+			.getBindingsFromShape<TLArrowBinding>(shape, 'arrow')
+			.every((binding) => selected.has(binding.toId))
+	})
+}
```
[the remark this answers is not in the document]
Comment (m31-galaxy) [reply packages/tldraw/src/lib/utils/frames/frames.ts:62]: Fixed in 76db47a: locked shapes are dropped first and each arrow's target is checked against that unlocked set rather than the raw selection. Arrows cannot bind to arrows, so a single pass is enough. Added a shortcut test and a menu item test for a locked target plus its bound arrow; both failed before the change.

[the remark this answers is not in the document]
Comment (m31-galaxy) [reply packages/tldraw/src/lib/utils/frames/frames.ts:62]: Fixed in 76db47a: the helper now checks bound targets against the shapes that survive the lock filter rather than the raw selection, so the arrow is dropped along with its locked target and the menu item stays hidden. Covered by new shortcut and menu item tests that failed before the change.

**Review by mimecuvalo — APPROVED** [review]
pre-approving but one edge case to address before landing: framing a single degenerate-bounds shape will throw (e.g. a horz arrow)

**Comment by m31-galaxy** [comment]
> pre-approving but one edge case to address before landing: framing a single degenerate-bounds shape will throw (e.g. a horz arrow)

On it!

**Comment by m31-galaxy** [comment]
Should be fixed with the latest commit, adding to merge queue now
